v2/http-middleware/ratelimit/ratelimit.go

```diff
@@ -18,87 +17,90 @@ type Store interface {
 	Disconnect() error
 }
 
-type EndpointConfig struct {
-	Path    Request
-	Configs []Config
-}
-
-type Config struct {
+type Limit struct {
 	RequestPerMinute int
-	GetKeyFunc       func(req *http.Request) (string, error)
+	Key              string
 }
 
 type Request struct {
-	Method string
-	Path   string
+	Method       string
+	PathTemplate string
 }
 
 type Limiter struct {
 	store   Store
-	configs map[Request][]Config
+	configs map[Request]limitGenerator
 }
 
-func CreateLimiter(s Store) Limiter {
-	return Limiter{
-		store:   s,
-		configs: map[Request][]Config{},
-	}
+type limitGenerator func(*http.Request) ([]Limit, error)
+
+func (l *Limiter) SetStore(s Store) *Limiter {
+	l.store = s
+	return l
 }
 
-// The GetKeyFunc should return a key that will be stored
-// as sha256(key) + <current minute in the cache to limit the number of
-// request using that key.
+// The LimitGenerator function should return the rate limit with corresponding key
+// that should be used for the given path template
+// path.pathTemplate should be the pathtemplate used to route the request
 //
 // If you give multiple configs for 1 endpoint. The most restrictive one will apply
-// The algorithm is inspired from: https://redislabs.com/redis-best-practices/basic-rate-limiting/
-func (s *Limiter) Configure(config EndpointConfig) {
-	s.configs[config.Path] = config.Configs
+func (s *Limiter) Configure(path Request, gen limitGenerator) {
+	if s.configs == nil {
+		s.configs = map[Request]limitGenerator{}
+	}
+
+	s.configs[path] = gen
 }
 
+// Rate limiting middleware, you can configure 1 or many limits for each path template using a limitGenerator
+// The algorithm is inspired from: https://redislabs.com/redis-best-practices/basic-rate-limiting/
+//
+// The key will be stored in clear text in the cache. If the key contains personal data please consider hashing the key
 func (s *Limiter) Middleware() mux.MiddlewareFunc {
 	if s.store == nil {
 		panic("store is not configured")
 	}
-	hasher := sha256.New()
 
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-			_, span := util.StartSpanNoRoot(req.Context(), "RateLimitMiddleware/Handler")
-			defer span.End()
+			ctx, span := util.StartSpanNoRoot(req.Context(), "RateLimitMiddleware/Handler")
 
 			now := time.Now()
 
-			// TODO: handle errors maybe allow access if we get error here?
-			cfgs, ok := s.configs[Request{Method: req.Method, Path: req.URL.Path}]
+			pathTemplate, err := mux.CurrentRoute(req).GetPathTemplate()
+			if err != nil {
+				log.WithTracing(ctx).WithError(err).Error("failed to get route template")
```

Comment:
Maybe make it more clear on what route template you are talking about. Since someone might think of other "routes"